airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/message/DestinationMessageQueue.kt

```diff
@@ -48,6 +49,7 @@ data class StreamCompleteWrapped(
  * This maximum is expected to be low, as the assumption is that data will be spooled to disk as
  * quickly as possible.
  */
+@Secondary
```

Comment:
There are two options for getting "deferential default".

One is to use `@Secondary`. The other is to put `@DefaultImplementation(SomeClass)` on the *interface.* @jdpgrailsdev recommended the latter, but I'm pretty sure it breaks the dependency tracing on the Micronaut plugin. @Secondary seems just as good for defaults?

Reply:
`@Secondary` or `@Primary` are fine to use so long as there are only two possible candidates.  If you end up in a situation with more than two candidates that need to be selected from, you will want to use a combination of `@Requires` to conditionally create the bean and `@Replaces` to ensure that conditionally selected candidate wins over any others that may also be conditionally created during execution.  See  https://docs.micronaut.io/latest/guide/#qualifiers (Primary and Secondary Beans subsection)  and https://docs.micronaut.io/latest/guide/#replaces for more information.